src/view/key_val/row.rs

```diff
@@ -67,16 +89,30 @@ mod imp {
         ) {
             match pspec.name() {
                 "key-val" => obj.set_key_val(value.get().unwrap_or_default()),
+                "key-label" => obj.set_key_label(value.get().unwrap_or_default()),
+                "value-label" => obj.set_value_label(value.get().unwrap_or_default()),
                 _ => unimplemented!(),
             }
         }
 
         fn property(&self, obj: &Self::Type, _id: usize, pspec: &glib::ParamSpec) -> glib::Value {
             match pspec.name() {
                 "key-val" => obj.key_val().to_value(),
+                "key-label" => obj.imp().key_label.borrow().to_value(),
+                "value-label" => obj.imp().value_label.borrow().to_value(),
                 _ => unimplemented!(),
             }
         }
+
+        fn constructed(&self, obj: &Self::Type) {
+            self.parent_constructed(obj);
+            obj.imp()
+                .key_entry
+                .set_property("placeholder-text", self.key_label.borrow().as_str());
+            obj.imp()
+                .value_entry
+                .set_property("placeholder-text", self.value_label.borrow().as_str());
```

Comment:
As we delegate to `Entry::placeholder_text` to write the property, we watch for changes here to notify about our property.

```suggestion
            self.key_entry
                .connect_placeholder_text_notify(clone!(@weak obj => move |_| {
                    obj.notify("key-label");
                }));
            self.key_entry
                .connect_placeholder_text_notify(clone!(@weak obj => move |_| {
                    obj.notify("value-label");
                }));
```